### ISSUE TITLE
[HOLD for payment 2024-11-13] [HOLD for payment 2024-11-11] [Workspace Feeds] Handle manual bank account verification before provisioning cards

### ISSUE THREAD

**Issue by mountiny** [Expensify/App#49472]
### Problem

In some cases, when policy admin is adding a VBBA, the bank account cannot be auto-approved, and we need to do manual checks to verify the bank account can be used for reimbursements and expensive cards. If that is the case, we need to make sure the user is not able to start issuing cards and they are aware a manual checks are happening. They receive a concierge message, which is easy to miss as currently they can just freely enter the Issue new card flow.

### Solution

When the VBBA account is added but is added to the waitlist for manual verification, let the user know in the Expensify cards modal that the card is still being manually verified and Concierge will notify them about the successful verification or if they need to provide any more details.

At the same time, disable the issue new card button, as until the bank account is verified, we cannot create new cards.

![image](https://github.com/user-attachments/assets/2101783c-22c9-489c-a23f-31c1ad4270f3)

<details><summary>Issue Owner</summary>Current Issue Owner: @sonialiap</details>

**Comment by trjExpensify** [comment]
Discussing [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1726757380005939?thread_ts=1725843290.225289&cid=C01GTK53T8Q) for the breadcrumbs.

**Comment by dubielzyk-expensify** [comment]
So shall we go for this then?

![image](https://github.com/user-attachments/assets/2101783c-22c9-489c-a23f-31c1ad4270f3)

**Comment by koko57** [comment]
I can help with this one 🙂

**Comment by trjExpensify** [comment]
Thanks! ❤️

**Comment by koko57** [comment]
For what bankAccount property should I look for to display the text and the buttons conditionally?

cc @sonialiap @mountiny

**Comment by koko57** [comment]
ok, UI is done, I drafted the PR, now I'm waiting for confirmation on this condition to display the buttons and the text

**Comment by mountiny** [comment]
Commented in the PR

**Comment by koko57** [comment]
@mountiny I followed the steps you shared to get account in VERIFIED state, but I don't have this account in bankAccountList so I'm still having Continue Modal displayed and "Finish Setup" as the button text
bc of this line `const isSetupUnfinished = isEmptyObject(bankAccountList) && reimbursementAccountStatus && reimbursementAccountStatus !== CONST.BANK_ACCOUNT.STATE.OPEN;`

<img width="1116" alt="Screenshot 2024-09-23 at 13 38 57" src="https://github.com/user-attachments/assets/d3ac7918-d4c5-4250-9764-b6caaa72a5e2">

**Comment by koko57** [comment]
So I guess I need to figure out another condition

**Comment by mountiny** [comment]
@koko57 ok, so you have the `private_expensifyCardSettings` NVP locally, and it only has the paymentAccountID linked, but that bank account is not in Onyx? Or what is the state of the Onyx variables after you added the bank account?

It seems like it's not exactly the same thing as what @ishpaul777 ran into as they had the card "provisioned" after adding the  bank account

**Comment by koko57** [comment]
no no no - I don't have paymentAccountID yet. We have it after we choose the bank account from the list. And after finishing the VBBA flow we don't have the bank account in the VERIFIED state yet in the list, so I don't have anything to choose from. If I had another bank account (with OPEN state) only the old one would be displayed

**Comment by mountiny** [comment]
I am just confused, how they could set up the card program then if they could not select the verifying bank account from the list 😕

**Comment by koko57** [comment]
not sure 🤷🏻‍♀️

**Comment by mountiny** [comment]
@ishpaul777 can you please copy here the requestID of the failed api call?

**Comment by koko57** [comment]
@mountiny and we have another problem - the object in bankAccountList doesn't have the state property right after adding a bank - so this info is not coming with `AcceptACHContractForBankAccount`. But after going to the wallet we get it with `OpenPaymentsPage`. I was able to add an account (with Open state, but this info was in achData in reimbursementAccount), then as the bankAccountList was not emty I was able to choose a bank account and go to the issue card flow and issue new Virtual Card. But as I don't have the state in the bankAccountList object see how the modal below the RHP shows "Verification in progress" and after I end the flow I don't see the buttons...

<img width="930" alt="Screenshot 2024-09-23 at 17 37 46" src="https://github.com/user-attachments/assets/f5c2b3f2-f24c-4b3f-ae5b-761ffd2a9e04">
<img width="1234" alt="Screenshot 2024-09-23 at 17 37 17" src="https://github.com/user-attachments/assets/97bc85f2-1bf9-405e-abd3-b6af51186681">


here's what I get from `AcceptACHContractForBankAccount`
<img width="591" alt="Screenshot 2024-09-23 at 17 36 52" src="https://github.com/user-attachments/assets/6857b310-0280-46e4-8cc2-fefd895f0b61">
<img width="694" alt="Screenshot 2024-09-23 at 17 36 30" src="https://github.com/user-attachments/assets/ca6e37cc-76bd-4b4a-ae15-cedbedf4b883">

the state is only in achData

**Comment by koko57** [comment]
@ishpaul777 could you please show what's sent with the request? I followed your steps but I was able to issue a card. The only difference was that I haven't changed the card type during the flow, maybe some data was lost after this change?

**Comment by koko57** [comment]
@ishpaul777 and for a new account and a new workspace - do you have the same problem?

**Comment by koko57** [comment]
and adding to my [earlier comment](https://github.com/Expensify/App/issues/49472#issuecomment-2367842453) - when we go to the Wallet page OpenPaymentsPage returns bankAccountList like this:
<img width="559" alt="Screenshot 2024-09-23 at 17 59 17" src="https://github.com/user-attachments/assets/cf244203-15f8-4c7c-8504-1cc238af29ba">
<img width="559" alt="Screenshot 2024-09-23 at 17 59 09" src="https://github.com/user-attachments/assets/6a34207a-a92c-4792-be65-3bd5595d1e41">

**Comment by koko57** [comment]
@ishpaul777 and for this account that doesn't work, do you remember how long ago it was when you enabled Expensify Card? If I remember correctly that we had similar problem when we didn't have EnablePolicyCompanyCards applied in the app.

**Comment by koko57** [comment]
@ishpaul777 and on this old policy - do you have workspaceAccountID in the policy object?
 I guess that areExpensifyCardEnabled is properly set to true.

**Comment by koko57** [comment]
thank you @nkuoch!

**Comment by koko57** [comment]
@ishpaul777 can you show how's the bankAccountList for the account that the flow doesn't work looking like?

**Comment by koko57** [comment]
@trjExpensify @mountiny so what's the final decision on this one?

**Comment by mountiny** [comment]
I don't think we got to the bottom of this one yet.

**Comment by mountiny** [comment]
Started a thread [here](https://expensify.slack.com/archives/C036QM0SLJK/p1727361426225429)

**Comment by mountiny** [comment]
Auth PR was merged, we need web PR will come off hold soon https://github.com/Expensify/Web-Expensify/pull/43767

**Comment by trjExpensify** [comment]
^^ Web PR is merged, awaiting deploys.

**Comment by sonialiap** [comment]
Deployed last night

**Comment by mountiny** [comment]
The waitlist NVP is now ready to be used. This nvp is sent to the user who was setting up the bank account, so not to all policy admins. Only the owner of the bank account.

**Comment by mountiny** [comment]
nvp_expensify_onCardWaitlist is the name of the nvp

**Comment by koko57** [comment]
I'll be working on this after https://github.com/Expensify/App/issues/50699, which I hopefully finish shortly

**Comment by koko57** [comment]
while working on this issue and some Company Cards issues I've noticed that the skeletons for both empty state pages are different - I know that their right side elements should differ (as we have two in EC and one in CC) but the layout and spacing looks different. Just wanted to let you know and ask if it's ok?

<img width="1261" alt="Screenshot 2024-10-30 at 17 13 19" src="https://github.com/user-attachments/assets/5d227dcd-cd83-4383-a212-8cc724e6ce78">
<img width="1259" alt="Screenshot 2024-10-30 at 17 13 13" src="https://github.com/user-attachments/assets/64f0d597-ee58-41aa-becb-953ceae44cd8">


cc @shawnborton @dubielzyk-expensify

**Comment by DylanDylann** [comment]
I believe that we need to put the modal in the middle

**Comment by koko57** [comment]
@DylanDylann it's not about the modals, but the skeletons under the modals. The modal is centered in relation to skeletons, so no need to change that

**Comment by mountiny** [comment]
thanks for raising, @Expensify/design which one should we use?

**Comment by dannymcclain** [comment]
The bottom one is correct. We recently updated the spacing for all those skeleton rows (from 12px to 8px) and I believe the top one is just a little outdated.

One thing though, I feel like you should be able to see more of the skeleton rows behind the modal? Here's what that screen looks like in Figma:
![image](https://github.com/user-attachments/assets/661d2be1-8bbe-42da-8247-ff8be0055705)

The skeleton rows seem to extend a lot further down than in the screenshot above.

**Comment by shawnborton** [comment]
Agree with both points from Danny there.

**Comment by DylanDylann** [comment]
@koko57 Waiting for you to raise PR here, right?

**Comment by koko57** [comment]
ok, we can do it here - but I'm ooo till Monday, so will take care of it then 🙂

**Comment by DylanDylann** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR: **New Feature**
 [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **NA**
 [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 [@DylanDylann] Determine if we should create a regression test for this bug. **Yes**
 [@DylanDylann] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


Regression Test Proposal

1. Go to a workspace, enable Expensify Card in More Features
2. Go to Expensify Card, click Issue New Card, add a new bank account if you don't have any
3. Choose the bank account from the list
4. After successful verification you should the Empty State screen
5. Repeat steps 1-3, choose the same bank account as in the previous workspace
6. You should see the Empty State screen and the buttons "Issue Card" and "Settings" should not be displayed.

Do we agree 👍 or 👎

**Comment by mountiny** [comment]
@DylanDylann @koko57 I dont think that test covers exactly what we need. We should check for the different copy on the modal for the second time

**Comment by DylanDylann** [comment]
@mountiny Thanks. I updated

**Comment by mountiny** [comment]
$250 to @DylanDylann

**Comment by sonialiap** [comment]
Payment summary:
- @DylanDylann $250 - offer sent

https://www.upwork.com/ab/applicants/1857030956887168073/hired

**Comment by DylanDylann** [comment]
@sonialiap Accepted, TY!

**Comment by sonialiap** [comment]
Paid 🎉
